### PR TITLE
Update ContactMap.match() and add naive method

### LINKED ISSUE #108: Structure maps with 0 contacts make the match function crash
## General Information
Structure maps with 0 contacts make the match function crash. It should go on with 0 TPs and 0 FNs.

- ConKit version: 0.13.2
- Python version:  >= 3.6
- Environment (if applicable): Anaconda

## Example

A minimal example to reproduce the error:
```
from conkit import io as ckio

# Parse input files
seq = ckio.read('mypath/myseq.fasta', 'fasta')[0]
strmap = ckio.read('mypath/mypdb.pdb', 'pdb')[0]
conpred = ckio.read('mypath/myconpred.psicov', 'psicov')

# Set sequence in contact prediction map
conpred.sequence = seq
conpred.set_sequence_register()

# Remove contacts from structure map
for contact in reversed(strmap):
    strmap.remove(contact.id)

# Match both
conpred.match(strmap, add_false_negatives=True, inplace=True)
```

## Traceback

The Python traceback
```
Traceback (most recent call last):

  File "/tmp/ipykernel_479769/2084760485.py", line 1, in <module>
    conpred.match(cmap, add_false_negatives=True, inplace=True)

  File "~/anaconda3/lib/python3.7/site-packages/conkit/core/contactmap.py", line 804, in match
    contact_map2_keymap = ContactMap._create_keymap(contact_map2, altloc=True)

  File "~/anaconda3/lib/python3.7/site-packages/conkit/core/contactmap.py", line 1126, in _create_keymap
    return list(zip(*contact_map_keymap_sorted))[1]

IndexError: list index out of range
```

### PR DESCRIPTION
**Pull request by FilomenoSanchez**
This fixes #108

### CODE REVIEW

**Comment by FilomenoSanchez** [comment]
@jjavier-bm, this includes the new method `conkit.ContactMap.match_naive` which can be used to match contact maps when no sequence alignment is necesary. Also, this fixes issue #108, now if the reference contact map is empty all predicted contacts are set to FP, or FN if the opposite is true.

**Review by jjavier-bm — APPROVED** [review]
All good. Thanks!

**Comment by FilomenoSanchez** [comment]
@hlasimpk in this update a method `match_naive` has been added to `conkit.core.ContactMap`. With this method, contact maps can be matched without any sequence alignment -so it assumes residue numbering is equivalent in both maps. It's a small update, let me know what you think!

File: conkit/core/contactmap.py [hunk 1]
```diff
@@ -682,9 +696,60 @@ def discard(*args, **kwargs):
...
+        -------
+        :obj:`~conkit.core.contactmap.ContactMap`
+            :obj:`~conkit.core.contactmap.ContactMap` instance, regardless of inplace
+
+        """
+        contact_map1 = self._inplace(inplace)
+        if match_other:
+            contact_map2 = other._inplace(inplace)
+        else:
+            contact_map2 = other._inplace(False)
```
Comment (hlasimpk) [inline conkit/core/contactmap.py:730]: If inplace is False, match_other doesn't actually do anything. Is this intended?

Comment (FilomenoSanchez) [reply conkit/core/contactmap.py:730]: Yes I agree this is a bit strange but this reproduces the behaviour of the original `conkit.core.ContactMap.match()` function so I left it as it is.

**Pull request merged**